src/core/public/saved_objects/saved_objects_client.ts

```diff
@@ -207,10 +222,53 @@ export class SavedObjectsClient {
     { leading: false }
   );
 
+  /**
+   * Throttled processing of resolve requests into bulk requests at 100ms interval
+   */
+  private processBatchResolveQueue = throttle(
+    async () => {
+      const queue = [...this.batchResolveQueue];
+      this.batchResolveQueue = [];
+
+      try {
+        const objectsToFetch = getObjectsToFetch(queue);
+        const { resolved_objects: savedObjects } = await this.performBulkResolve(objectsToFetch);
+
+        queue.forEach((queueItem) => {
+          const foundObject = savedObjects.find((resolveResponse) => {
+            return (
+              resolveResponse.saved_object.id === queueItem.id &&
+              resolveResponse.saved_object.type === queueItem.type
+            );
+          });
+
+          if (foundObject) {
+            // multiple calls may have been requested the same object.
+            // we need to clone to avoid sharing references between the instances
+            queueItem.resolve(this.createResolvedSavedObject(cloneDeep(foundObject)));
+          } else {
+            queueItem.resolve(
```

Comment:
Good question, I just mimicked the implementation of the existing `processBatchGetQueue` and it appears to be a fail-safe in case Kibana failed to respond with a result for the requested object.
That Shouldn't Ever Happen™ so I doubt much thought was put into the original implementation when it was written what appears to be several years ago. I'm just hesitant to make any changes that aren't absolutely necessary.

Reply:
I agree with Joe. Even if terrible design, this is the current behavior of the bulkGet queue, so mimicking it for this additional queue makes sense imho. At one point we'll like to eventually get rid of this terrible, terrible thing, but for the time being, having both `get` and its 'pseudo replacement' `resolve` with the same behavior feels like the safest thing to do.

Reply:
👍 I'm on board with consistency here. Thanks